test/integration/enterprise_test.go

```diff
@@ -0,0 +1,55 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	kongaddon "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong"
+	metallbaddon "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/metallb"
+	environment "github.com/kong/kubernetes-testing-framework/pkg/environments"
+)
+
+func TestKongEnterpriseWithDBLessMode(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*10)
+	defer cancel()
+
+	t.Log("configuring the testing environment")
+	metallb := metallbaddon.New()
+	kong := kongaddon.NewBuilder().WithEnterprise(kongaddon.DBLESS).WithImage(kongaddon.EnterpriseImageRepo, kongaddon.EnterpriseImageTag).Build()
+	builder := environment.NewBuilder().WithAddons(kong, metallb)
+
+	t.Log("building the testing environment and Kubernetes cluster")
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+
+	t.Logf("setting up the environment cleanup for environment %s and cluster %s", env.Name(), env.Cluster().Name())
+	defer func() {
+		t.Logf("cleaning up environment %s and cluster %s", env.Name(), env.Cluster().Name())
+		require.NoError(t, env.Cleanup(ctx))
+	}()
+
+	t.Log("waiting for environment to be ready")
+	require.NoError(t, <-env.WaitForReady(ctx))
+
+	t.Logf("verifying that the kong proxy service %s gets provisioned an IP address by metallb", kongaddon.DefaultProxyServiceName)
+	proxyURL, err := kong.ProxyURL(ctx, env.Cluster())
+	require.NoError(t, err)
+	require.NotNil(t, proxyURL)
+
+	t.Logf("found url %s for proxy, now verifying it is routable", proxyURL)
+	httpc := http.Client{Timeout: time.Second * 3}
+	require.Eventually(t, func() bool {
+		resp, err := httpc.Get(proxyURL.String())
+		if err != nil {
+			return false
+		}
+		defer resp.Body.Close()
+		return resp.StatusCode == http.StatusNotFound
+	}, time.Minute*1, time.Second*1)
```

Comment:
So far this just deploys a cluster with dbless enabled but only verifies that the proxy came up. We should expand these tests to include some minimal verification that the enterprise proxy is actually in use. I don't have too many suggestions, but I think simply identifying some really basic enterprise functionality (perhaps an admin api route that's only available when enterprise is enabled) and doing a small test of that would make this test have more impact.

Reply:
In CI, current integration tests can satisfy 2.0 KIC and enterprise. Following on, we can add more enterprise license featured test cases, such as workspace, and RBAC-protected instance based on my current understanding.

Reply:
created follow up issues https://github.com/Kong/kubernetes-ingress-controller/issues/1785

Reply:
Sorry, I don't see this follow up as sufficient to resolve my concerns about testing so I'm marking this as unresolved.

I believe we should actually test and verify the functionality we are adding with this PR here in this repo, doing the testing downstream means we can make a release that we wont know has broken enterprise support and we wont know it until we try to include that release in downstream which is not how this should work.

We don't need to thoroughly test all enterprise features, upstream kong does that. What we do need to do is verify that enterprise features are in fact available which some simple tests would be sufficient for.

Reply:
f47f2681a07ca1cb9b759eb0011b81dd9eb886fb